llvm/lib/Target/NVPTX/NVPTXIntrinsics.td

```diff
@@ -7078,3 +7079,713 @@ defm INT_SET_MAXNREG_DEC : SET_MAXNREG<"dec", int_nvvm_setmaxnreg_dec_sync_align
 } // isConvergent
 
 def INT_EXIT : NVPTXInst<(outs), (ins), "exit;", [(int_nvvm_exit)]>;
+
+def : Pat<(int_nvvm_tex_1d_v4f32_s32 i64:$t, i64:$s, i32:$x),
```

Comment:
All these patterns can probably be multiclassed or integrated into instruction definitions.
I'm not ready to do that, but I can add a TODO.

Reply:
In the end I did it.